io/src/Register.hpp

```diff
@@ -28,7 +28,7 @@
 #include <boost/serialization/serialization.hpp>
 #include <boost/serialization/nvp.hpp>
 
-#if __cplusplus >= 201103L
+#if SICONOS_CXXVERSION >= 201103L
```

Comment:
Note: This was the only one I wasn't sure to do with in terms of feature detection, so I at least changed the comparison with __cplusplus for SICONOS_CXXVERSION.